internal/scheduler/nodedb/nodedb.go

```diff
@@ -155,6 +160,18 @@ func (nodeDb *NodeDb) String() string {
 	return sb.String()
 }
 
+func (nodeDb *NodeDb) NumNodes() int {
```

Comment:
These functions are a bit weird because they have mutexes, which implies that the dat can be modified by another thread, but if so then they don't seem that useful?
e.g. if I do:
```
if nodeDb.NumNodes() > 0 {
   // Do something that requires at least one node
}
```
It's not guaranteed to work because NumNodes could have changed between my check and the actual code. I think in practise this doesn't matter because right now nodeDb is only accessed from a single thread, but I think it makes the mutex unnecessary.  More generally, to make this work in a multi thread env, we'd need to change the nbodedb so that all functions operate off a snapshot of data, so you can guarantee repeatable read.

Reply:
You're right. The NodeDb isn't thread-safe anyway. I'll leave the lock anyway, since it at least protects against races during initialisation.